galsim/vonkarman.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) 2012-2017 by the GalSim developers team on GitHub
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+"""@file base.py
+This file implements the von Karman atmospheric PSF profile.
+"""
+
+import numpy as np
+
+import galsim
+
+from . import _galsim
+from .gsobject import GSObject
+
+
+class VonKarman(GSObject):
+    """Class describing a von Karman surface brightness profile, which represents a long exposure
+    atmospheric PSF.  The difference between the von Karman profile and the related Kolmogorov
+    profile is that the von Karman profile includes a parameter for the outer scale of atmospheric
+    turbulence, which is a physical scale beyond which fluctuations in the refractive index stop
+    growing, typically between 10 and 100 meters.  Quantitatively, the von Karman phase fluctuation
+    power spectrum is proportional to
+
+        (f^2 + L0^-2)^(-11/6)
+
+    where f is a spatial frequency and L0 is the outer scale in meters.  The Kolmogorov power
+    spectrum proportional to f^(-11/3) is recovered as L0 -> infinity,
+
+    For more information, we recommend the following references:
+
+        Martinez et al.  2010  A&A  vol. 516
+        Conan  2008  JOSA  vol. 25
+
+    Notes
+    -----
+
+    If one blindly follows the math for converting the von Karman power spectrum into a PSF, one
+    finds that the PSF contains a delta-function at the origin with fractional flux of
+
+        exp(-0.5*0.172*(r0/L0)^(-5/3))
+
+    In almost all cases of interest this evaluates to something tiny, often on the order of 10^-100
+    or smaller.  By default, GalSim will ignore this delta function entirely.  If for some reason
+    you want to keep the delta function, though, then you can pass the do_delta=True argument to the
+    VonKarman initializer.
+
+    @param lam               Wavelength in nanometers
+    @param r0                Fried parameter in meters.
+    @param L0                Outer scale in meters.  [default: np.inf]
```

Comment:
So our default makes von Karman equivalent to Kolmogorov?  Hmmm.  I guess choosing some random non-infinite value between 10-100 meters seems too arbitrary as a default?

Reply:
Yeah, I'm not sure what the best value is here.  I did go ahead and change this to 25.0 m though, which makes it consistent with the default value in `AtmosphericScreen`.